src/System.Runtime.Caching/src/System/Runtime/Caching/ObjectCache.cs

```diff
@@ -0,0 +1,129 @@
+        {
+            return (AddOrGetExisting(key, value, policy, regionName) == null);
+        }
+
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification = "This is assembly is a special case approved by the NetFx API review board")]
+        public abstract object AddOrGetExisting(string key, object value, DateTimeOffset absoluteExpiration, string regionName = null);
+        public abstract CacheItem AddOrGetExisting(CacheItem value, CacheItemPolicy policy);
+
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification = "This is assembly is a special case approved by the NetFx API review board")]
+        public abstract object AddOrGetExisting(string key, object value, CacheItemPolicy policy, string regionName = null);
+
+        [SuppressMessage("Microsoft.Naming", "CA1716:IdentifiersShouldNotMatchKeywords", Justification = "The name best represents an operation on a cache")]
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification = "This is assembly is a special case approved by the NetFx API review board")]
+        public abstract object Get(string key, string regionName = null);
+
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification = "This is assembly is a special case approved by the NetFx API review board")]
+        public abstract CacheItem GetCacheItem(string key, string regionName = null);
+
+        [SuppressMessage("Microsoft.Naming", "CA1716:IdentifiersShouldNotMatchKeywords", Justification = "The name best represents an operation on a cache")]
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification = "This is assembly is a special case approved by the NetFx API review board")]
+        public abstract void Set(string key, object value, DateTimeOffset absoluteExpiration, string regionName = null);
+
+        [SuppressMessage("Microsoft.Naming", "CA1716:IdentifiersShouldNotMatchKeywords", Justification = "The name best represents an operation on a cache")]
+        public abstract void Set(CacheItem item, CacheItemPolicy policy);
+
+        [SuppressMessage("Microsoft.Naming", "CA1716:IdentifiersShouldNotMatchKeywords", Justification = "The name best represents an operation on a cache")]
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification = "This is assembly is a special case approved by the NetFx API review board")]
+        public abstract void Set(string key, object value, CacheItemPolicy policy, string regionName = null);
+
+        //Get multiple items by keys
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification = "This is assembly is a special case approved by the NetFx API review board")]
```

Comment:
you can remove all CA1026 as we don't enforce that rule. We use default params al ot.